### PR TITLE
feat: quantity_units_new_fields

### LINKED ISSUE #9377: Add product_quantity_unit and serving_quantity_unit fields
The "quantity" and "serving_size" fields are free text fields. They are converted to a quantity in g or ml and the result is stored in "product_quantity" and "serving_quantity". We should add new fields that indicate if those normalized _quantity fields are in g or ml.

### PR DESCRIPTION
**Pull request by benbenben2**
### What

Adding new fields:
- product_quantity_unit
- serving_quantity_unit

Related to this PR from @stephanegigandet: #9573. 

### Screenshot
Screenshot_20231231_114016

### Related issue(s) and discussion
- Fixes #9377

### CODE REVIEW

File: lib/ProductOpener/Food.pm [hunk 1]
```diff
@@ -2113,15 +2113,21 @@ sub compute_serving_size_data ($product_ref) {
...
 		}
+		my $product_quantity_unit = extract_standard_unit($product_ref->{quantity});
+		if (defined $product_quantity_unit) {
+			$product_ref->{product_quantity_unit} = $product_quantity_unit;
+		}
 	}
 
 	if ((defined $product_ref->{serving_size}) and ($product_ref->{serving_size} ne "")) {
 		$product_ref->{serving_quantity} = normalize_serving_size($product_ref->{serving_size});
+		$product_ref->{serving_quantity_unit} = extract_standard_unit($product_ref->{serving_size});
```
Comment (stephanegigandet) [inline lib/ProductOpener/Food.pm:2130]: Shouldn't there be a check that the unit is not undef, as you did for product_quantity_unit?

**Review by stephanegigandet — APPROVED** [review]
Looks good, one minor comment

File: lib/ProductOpener/Food.pm (hunk 1 above)
Comment (benbenben2) [reply lib/ProductOpener/Food.pm]: Good point

**Comment by benbenben2** [comment]
Thank you, for the review

**Comment by hrabkin** [comment]
I tried to rely on this property but after checking some products I don't see it at all for e.g. in this product https://world.openfoodfacts.org/api/v3/product/5903978396357.json its missing, also there's unit in quantity in liters but product_quantity in millilitres:
```
"product_quantity": "1000",
"quantity": "1l",
```

It's pretty hard to make a good high quality product on such unstable data structure

**Comment by benbenben2** [comment]
Hi @hrabkin 

You have to wait for the next release to be able to see the changes.

product_quantity => "The size in g or ml for the whole product. It's a normalized version of the quantity field."
quantity => "Quantity and Unit."
(refer to docs/api/ref/schemas/product_base.yaml)

So typically, **product_quantity** extract the number from **quantity** and normalize it in either g or ml. So 1L in **quantity** would be 1000 (mL) in **product_quantity**. **quantity** is input from contributors on the app or the website, **product_quantity** is done automatically based on quantity. The puprose of this PR was to also have the field of the extracted unit, "mL" in the example I just gave. Hope this help. 

Which make me think: @stephanegigandet, @alexgarel, should I have added "product_quantity_unit" and "serving_quantity_unit" in the docs as well? => I did a PR for that: https://github.com/openfoodfacts/openfoodfacts-server/pull/9650

**Pull request merged**